codegen/java-gen/src/main/scala/com/lightbend/akkasls/codegen/java/ActionServiceSourceGenerator.scala

```diff
@@ -78,11 +78,12 @@ object ActionServiceSourceGenerator {
         implSourcePath,
         actionSource(service).getBytes(Charsets.UTF_8)
       )
-
-      List(implSourcePath, interfaceSourcePath, providerSourcePath, handlerSourcePath)
-    } else {
-      List(interfaceSourcePath, providerSourcePath, handlerSourcePath)
     }
+    // We return implSourcePath here even when we didn't just generate it, because
+    // otherwise the incremental compiler can't seem to find it. I'm not entirely confident
+    // this is the right way to fix that, but it seems to work. Let's revisit when we find a
+    // problem with this approach.
```

Comment:
I think I noted the same problem in the sbt codegen test trixery

Reply:
I don't understand why that would fix anything. In the two places where we call the generate method, we don't really use the returned Seq of Paths. 

The maven plugin does print out what was generated and in that case it may give the impression that the Main and implementation classes got regenerated, which is not true.